presto-hive/src/main/java/io/prestosql/plugin/hive/util/Statistics.java

```diff
@@ -99,12 +99,18 @@ public static HiveBasicStatistics reduce(HiveBasicStatistics first, HiveBasicSta
 
     public static Map<String, HiveColumnStatistics> merge(Map<String, HiveColumnStatistics> first, Map<String, HiveColumnStatistics> second)
     {
-        // only keep columns that have statistics for both sides
-        Set<String> columns = intersection(first.keySet(), second.keySet());
-        return columns.stream()
+        verify(first != null, "first is null");
+        verify(second != null, "second is null");
+
+        Set<String> columns = union(first.keySet(), second.keySet());
+        HiveColumnStatistics emptyStats = HiveColumnStatistics.builder().build();
+
+        ImmutableMap<String, HiveColumnStatistics> collect = columns.stream()
                 .collect(toImmutableMap(
                         column -> column,
-                        column -> merge(first.get(column), second.get(column))));
+                        column -> merge(first.getOrDefault(column, emptyStats), second.getOrDefault(column, emptyStats))));
```

Comment:
That's not safe in general. It's safe to take empty stats when stats are missing only if you have other basis to conclude the table was empty. eg you could capture row count from basic table stats and do this only if it was 0